saleor/graphql/core/fields.py

```diff
@@ -66,3 +68,55 @@ def resolve_connection(cls, connection, default_manager, args, iterable):
         connection.iterable = iterable
         connection.length = _len
         return connection
+
+
+class FilterInputConnectionField(DjangoConnectionField):
+    def __init__(self, *args, **kwargs):
+        filters_name = kwargs.get('filters_name', 'filters')
+        self.filters_obj = kwargs.get(filters_name)
+        self.filterset_class = None
+        if self.filters_obj:
+            self.filterset_class = self.filters_obj.filterset_class
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def connection_resolver(
+            cls, resolver, connection, default_manager, max_limit,
+            enforce_first_or_last, filterset_class, root, info, **args):
+
+        # Disable `enforce_first_or_last` if not querying for `edges`.
+        values = [
+            field.name.value
+            for field in info.field_asts[0].selection_set.selections]
+        if 'edges' not in values:
+            enforce_first_or_last = False
+
+        filters_input = args.get('filters')
```

Comment:
I might be wrong, but shouldn't this actually use `filters_name` from the `__init__`? Isn't the `filters` name hardcoded?

Reply:
@maarcingebala  you are right